homeassistant/components/jewish_calendar/sensor.py

```diff
@@ -128,6 +128,8 @@ def get_state(self, after_shkia_date, after_tzais_date):
             return after_shkia_date.holiday_description
         if self._type == "omer_count":
             return after_shkia_date.omer_day
+        if self._type == "daf_yomi":
+            return hdate.HDate(today, diaspora=self._diaspora, hebrew=self._hebrew).daf_yomi
```

Comment:
Just return `date.daf_yomi` if you want it to change at midnight.

Reply:
It looks like the `date` variable is out of scope here. Were you suggesting modifying the method's signature to pass in `date` ?

Reply:
You're right, I changed that in a previous PR. :)
Maybe change the signature to add `after_midnight_date` which will get the date value.
I think it will make the code easier to read and understand.

Reply:
Note that you'll need to update the method in the `JewishCalendarTimeSensor` as well.

Reply:
I updated both